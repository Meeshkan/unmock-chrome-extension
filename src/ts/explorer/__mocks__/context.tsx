import * as React from "react";

const actions = {};
export const ExplorerActionsContext = React.createContext({ actions });
