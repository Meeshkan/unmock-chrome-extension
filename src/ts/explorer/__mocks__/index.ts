import { Labeled, Phase } from "../../state";
import { ExplorerState } from "../explorer";

export const mockActions = {
  triggerFetchSuccess: jest.fn(),
  triggerSetActiveUrl: jest.fn(),
  triggerInitializeStore: jest.fn(),
  triggerDownload: jest.fn(),
};

export const mockLabeled: Labeled = {
  "https://docs.readthedocs.io/en/stable/api/v2.html": {
    "/api/v2/project/": {
      GET: {},
      PUT: {},
    },
    "/api/v2/user/": {
      POST: {},
    },
  },
  "https://docs.readthedocs.io/en/stable/api/v1.html": {
    "/api/v1/project/": {
      GET: {},
      PUT: {},
    },
    "/api/v1/user/": {},
  },
};

export const mockExplorerState: ExplorerState = {
  data: {
    active: {
      activePath: [],
      phase: Phase.ADD_PATH,
    },
    labeled: mockLabeled,
  },
  isLoading: false,
};
