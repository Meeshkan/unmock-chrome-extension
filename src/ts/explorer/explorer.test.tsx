import * as React from "react";
import Explorer from "./explorer";
import { mockActions, mockExplorerState } from "./__mocks__";
import { shallow } from "enzyme";

const props = {
  state: mockExplorerState,
  actions: mockActions,
};

describe("Explorer", () => {
  test("renders correctly", () => {
    const wrapper = shallow(<Explorer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
