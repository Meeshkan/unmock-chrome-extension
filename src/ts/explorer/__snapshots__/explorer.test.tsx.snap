// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer renders correctly 1`] = `
<div>
  <h1>
    Welcome to Unmock API labeling explorer!
  </h1>
  <div>
    <ActiveStateComponent
      active={
        Object {
          "activePath": Array [],
          "phase": "Add path",
        }
      }
    />
    <LabeledUrlsComponent
      actions={
        Object {
          "triggerDownload": [MockFunction],
          "triggerFetchSuccess": [MockFunction],
          "triggerInitializeStore": [MockFunction],
          "triggerSetActiveUrl": [MockFunction],
        }
      }
      active={
        Object {
          "activePath": Array [],
          "phase": "Add path",
        }
      }
      labeled={
        Object {
          "https://docs.readthedocs.io/en/stable/api/v1.html": Object {
            "/api/v1/project/": Object {
              "GET": Object {},
              "PUT": Object {},
            },
            "/api/v1/user/": Object {},
          },
          "https://docs.readthedocs.io/en/stable/api/v2.html": Object {
            "/api/v2/project/": Object {
              "GET": Object {},
              "PUT": Object {},
            },
            "/api/v2/user/": Object {
              "POST": Object {},
            },
          },
        }
      }
    />
    <ExplorerButtonsComponent
      actions={
        Object {
          "triggerDownload": [MockFunction],
          "triggerFetchSuccess": [MockFunction],
          "triggerInitializeStore": [MockFunction],
          "triggerSetActiveUrl": [MockFunction],
        }
      }
      labeled={
        Object {
          "https://docs.readthedocs.io/en/stable/api/v1.html": Object {
            "/api/v1/project/": Object {
              "GET": Object {},
              "PUT": Object {},
            },
            "/api/v1/user/": Object {},
          },
          "https://docs.readthedocs.io/en/stable/api/v2.html": Object {
            "/api/v2/project/": Object {
              "GET": Object {},
              "PUT": Object {},
            },
            "/api/v2/user/": Object {
              "POST": Object {},
            },
          },
        }
      }
    />
  </div>
</div>
`;
